pkg/chart/chart.go

```diff
@@ -0,0 +1,68 @@
+// Copyright SUSE LLC.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chart
+
+import (
+	"github.com/Masterminds/log-go"
+	"gopkg.in/yaml.v2"
+	helmChart "helm.sh/helm/v3/pkg/chart"
+)
+
+// Dependency is a composite type of Helm's chart.Dependency
+type Dependency struct {
+	*helmChart.Dependency
+	// hypper specific
+	IsOptional bool `json:"-"`
+}
+
+// GetSharedDeps returns a *[] of all shared and optional dependencies in a
+// chart, read from annotations.
+func GetSharedDeps(c *helmChart.Chart, logger log.Logger) ([]*Dependency, error) {
```

Comment:
GetSharedDeps() is missing tests.
It would be nice to use pkg/action/action_test.go mocks for it, but, they aren't exported.